view/Requirements.php

```diff
@@ -892,24 +891,37 @@ public function includeInHTML($templateFile, $content) {
 						$commentTags[1] == '-->'
 					);
 
-				if($canWriteToBody) {
-					$content = substr($content,0,$p1) . $jsRequirements . substr($content,$p1);
+				if ($canWriteToBody) {
+					$content = substr($content, 0, $p1) . $jsRequirements . substr($content, $p1);
 				} else {
-					$content = preg_replace("/(<\/body[^>]*>)/i", $jsRequirements . "\\1", $content);
+					$replacements["/(<\/body[^>]*>)/i"] = $jsRequirements . "\\1";
 				}
 
 				// Put CSS at the bottom of the head
-				$content = preg_replace("/(<\/head>)/i", $requirements . "\\1", $content);
+				$replacements["/(<\/head>)/i"] = $requirements . "\\1";
 			} else {
-				$content = preg_replace("/(<\/head>)/i", $requirements . "\\1", $content);
-				$content = preg_replace("/(<\/head>)/i", $jsRequirements . "\\1", $content);
+				// Put CSS and Javascript together before the closing head tag
+				$replacements["/(<\/head>)/i"] = $requirements . $jsRequirements. "\\1";
 			}
+
+			// Replace everything at once (only once)
+			$content = preg_replace(array_keys($replacements), array_values($replacements), $content, 1);
```

Comment:
Maybe add `if($replacements)` here? If `$canWriteToBody and $this->write_js_to_body` are both set I think that replacements will be empty?